### PR TITLE
fix(slices): Rename partition to slice 

### PR DESCRIPTION
**Pull request by ayirr7**
`is_storage_partitioned` to `is_storage_sliced`

Calling the storage "partitioned" in the context of slicing can be confusing, so changing it to "sliced" since this will refer to storages that are associated with a storage set key that is sliced. Storage sets are still the level at which slicing is happening, but we want to be consistent with slicing config names in settings like `SLICED_STORAGES`

### CODE REVIEW

**Review by nikhars — COMMENTED** [review]
Let's rename the file snuba/datasets/partitioning.py to snuba/datasets/slicing.py. 

Let's also update the docs and rename partitioning to slicing. And change the name of the file to slicing.rst. https://github.com/getsentry/snuba/blob/master/docs/source/architecture/partitioning.rst to

File: snuba/datasets/pluggable_entity.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ def get_all_join_relationships(self) -> Mapping[str, JoinRelationship]:
     def get_query_pipeline_builder(self) -> QueryPipelineBuilder[ClickhouseQueryPlan]:
         from snuba.pipeline.simple_pipeline import SimplePipelineBuilder
 
-        if is_storage_partitioned(self.readable_storage.get_storage_key()):
+        if is_storage_sliced(self.readable_storage.get_storage_key()):
             assert (
                 self.partition_key_column_name is not None
```
Comment (lynnagara) [inline snuba/datasets/pluggable_entity.py:74]: `slice_key_column_name`?

**Review by lynnagara — COMMENTED** [review]
Thanks for this! 😍 for naming consistency everywhere

**Review by nikhars — APPROVED** [review]

**Comment by nikhars** [comment]
Please fix the docs build failure before merging.

**Pull request merged**